packages/module-source/src/transformSource.js

```diff
@@ -26,22 +26,31 @@ export const makeTransformSource = (makeModulePlugins, babel = null) => {
     const { sourceUrl, sourceMapUrl, sourceType, sourceMap, sourceMapHook } =
       sourceOptions;
 
-    const ast = parseBabel(source, { sourceType });
+    const ast = parseBabel(source, {
+      sourceType,
+      tokens: true,
+      createParenthesizedExpression: true,
+    });
 
     traverseBabel(ast, visitorFromPlugin(analyzePlugin));
     traverseBabel(ast, visitorFromPlugin(transformPlugin));
 
     const sourceMaps = sourceOptions.sourceMapHook !== undefined;
 
     const { code: transformedSource, map: transformedSourceMap } =
-      generateBabel(ast, {
-        sourceFileName: sourceMapUrl,
-        sourceMaps,
-        inputSourceMap: sourceMap,
-        retainLines: true,
-        compact: true,
-        verbatim: true,
-      });
+      generateBabel(
+        ast,
+        {
+          sourceFileName: sourceMapUrl,
+          sourceMaps,
+          inputSourceMap: sourceMap,
+          retainLines: true,
+          preserveFormat: true,
+          compact: true,
```

Comment:
@nicoloribaudo Reviewing https://github.com/babel/babel/pull/16708/files#diff-ca2bda59eec9c35846e26c1c6247759c92c26357b80d4dc881fcdaf12df1d7e6R35-R39, it seems like `compact: true` here should be causing your code to throw, so I suspect this isn’t exercising the new path in my local testing.

Reply:
Found it. Needed to swap out our fork `@agoric/babel-generator` back to `@babel/generator`.